android/MobiledgeXSDKDemo/matchingenginehelper/src/main/java/com/mobiledgex/matchingenginehelper/MatchingEngineHelper.java

```diff
@@ -243,56 +256,67 @@ public MatchingEngineHelper(final Builder builder) {
      */
     protected void initEdgeEventsConfigPrefs() {
         SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(mActivity);
-        onSharedPreferenceChanged(prefs, prefKeyLocationUpdatePattern);
-        onSharedPreferenceChanged(prefs, prefKeyLocationUpdateInterval);
-        onSharedPreferenceChanged(prefs, prefKeyLocationMaxNumUpdates);
-        onSharedPreferenceChanged(prefs, prefKeyFindCloudletEventTrigger);
-        onSharedPreferenceChanged(prefs, prefKeyLatencyTestType);
-        onSharedPreferenceChanged(prefs, prefKeyLatencyUpdatePattern);
-        onSharedPreferenceChanged(prefs, prefKeyLatencyTestTriggerMode);
-        onSharedPreferenceChanged(prefs, prefKeyLatencyUpdateInterval);
-        onSharedPreferenceChanged(prefs, prefKeyLatencyMaxNumUpdates);
-        onSharedPreferenceChanged(prefs, prefKeyLatencyTestPort);
-        onSharedPreferenceChanged(prefs, prefKeyLatencyThreshold);
-        onSharedPreferenceChanged(prefs, prefKeyPerfMarginSwitch);
-        onSharedPreferenceChanged(prefs, prefKeyAutoMigration);
+        onEdgeEventPreferenceChanged(prefs, prefKeyEnableEdgeEvents);
+        onEdgeEventPreferenceChanged(prefs, prefKeyOverrideDefaultConfig);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLocationUpdatePattern);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLocationUpdateInterval);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLocationMaxNumUpdates);
+        onEdgeEventPreferenceChanged(prefs, prefKeyFindCloudletEventTrigger);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyTestType);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyUpdatePattern);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyTestTriggerMode);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyUpdateInterval);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyMaxNumUpdates);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyTestPort);
+        onEdgeEventPreferenceChanged(prefs, prefKeyLatencyThreshold);
+        onEdgeEventPreferenceChanged(prefs, prefKeyPerfMarginSwitch);
+        onEdgeEventPreferenceChanged(prefs, prefKeyAutoMigration);
     }
 
-    public void startEdgeEvents() {
+    public void startEdgeEvents(boolean newConnection) {
+        if (!mEdgeEventsEnabled) {

```

Comment:
I don't see the full app context, but here are some public functions to avoid consistency issues:
public boolean isEnableEdgeEvents();
synchronized public void setEnableEdgeEvents(boolean enableEdgeEvents);

Reply:
This particular `mEdgeEventsEnabled` variable represents the app's preference value for "Enable Edge Events" which the user can turn on or off.